tools/database/src/block_iterators/mod.rs

```diff
@@ -0,0 +1,61 @@
+//! This module contains iterators that can be used to iterate over blocks in the database
+
+mod height_range;
+mod last_blocks;
+
+use std::rc::Rc;
+
+use near_chain::{Block, ChainStore};
+use near_primitives::types::BlockHeight;
+
+/// Iterate over blocks between two block heights.
+/// `from_height` and `to_height` are inclusive
+pub use height_range::BlockHeightRangeIterator;
+
+/// Iterate over the last N blocks in the blockchain
+pub use last_blocks::LastNBlocksIterator;
+
+/// Arguments that user can pass to a command to choose some subset of blocks
+pub struct CommandArgs {
```

Comment:
I put `CommandArgs` and the logic of creating a block iterator from them in a separate module because I wanted to make it possible to reuse it in another command. In the future we might want to implement another command that analyses a range of blocks, and it would be nice to be able to reuse the existing logic from `analyse-gas-usage`.

Reply:
I fully agree, my suggestion was to have separate methods for the different types of iterators. This way you can get rid of the validation of user input from the iterators library. It's only a suggestion and it's just fine as is if you like this more.

Reply:
> my suggestion was to have separate methods for the different types of iterators.

I think this is already provided by the `new` functions of each iterator. You can call `LastNBlocksIterator::new()` or `BlockHeightRangeIterator::new()` to create the desired iterator directly.